backend/backend.go

```diff
@@ -83,6 +83,24 @@ type Local interface {
 	Context(*Operation) (*terraform.Context, state.State, error)
 }
 
+// Creator can be implemented by backends to automatically create
+// necessary resources during `init`. The backend will be fully configured
```

Comment:
I wonder if the challenges you describe here are an argument that initializing resources for the backend shouldn't be part of Terraform's job and should rather be done out-of-band some other way.

With that said, with `RequiresCreate` here I suppose this is somewhat mitigated, since it gives Terraform an opportunity to describe what it's about to do, hopefully in a way that doesn't make it feel like Terraform is taking responsibililty for managing the created resources in the same sense it usually would.

I can definitely see the UX benefits of having `terraform init` be able to, say, create the "project" object on the backend if we detect it doesn't already exist... let's just make sure we take care with this UX to reduce confusion. e.g. I wouldn't want to see the word "resource" appear in the UI in this context, when you get there. :grinning: